pkg/providers/vsphere/vsphere.go

```diff
@@ -1152,11 +1156,35 @@ func (p *vsphereProvider) PostBootstrapDeleteForUpgrade(ctx context.Context) err
 	return nil
 }
 
-// PreCoreComponentsUpgrade staisfies the Provider interface.
+// PreCoreComponentsUpgrade satisfies the Provider interface.
 func (p *vsphereProvider) PreCoreComponentsUpgrade(
```

Comment:
Is there a method we can run after new nodes have been rolled out to delete this DS?

Reply:
Good idea. I can create a method to do this